shared/pkg/families/config.go

```diff
@@ -16,15 +16,48 @@
 package families
 
 import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/anchore/syft/syft/source"
+	kubeclarityConfig "github.com/openclarity/kubeclarity/shared/pkg/config"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+
+	"github.com/openclarity/vmclarity/api/models"
 	"github.com/openclarity/vmclarity/shared/pkg/families/exploits"
+	exploitsCommon "github.com/openclarity/vmclarity/shared/pkg/families/exploits/common"
+	exploitdbConfig "github.com/openclarity/vmclarity/shared/pkg/families/exploits/exploitdb/config"
 	"github.com/openclarity/vmclarity/shared/pkg/families/malware"
+	malwareconfig "github.com/openclarity/vmclarity/shared/pkg/families/malware/clam/config"
+	malwarecommon "github.com/openclarity/vmclarity/shared/pkg/families/malware/common"
 	misconfigurationTypes "github.com/openclarity/vmclarity/shared/pkg/families/misconfiguration/types"
 	"github.com/openclarity/vmclarity/shared/pkg/families/rootkits"
+	chkrootkitConfig "github.com/openclarity/vmclarity/shared/pkg/families/rootkits/chkrootkit/config"
+	rootkitsCommon "github.com/openclarity/vmclarity/shared/pkg/families/rootkits/common"
 	"github.com/openclarity/vmclarity/shared/pkg/families/sbom"
 	"github.com/openclarity/vmclarity/shared/pkg/families/secrets"
+	"github.com/openclarity/vmclarity/shared/pkg/families/secrets/common"
+	gitleaksconfig "github.com/openclarity/vmclarity/shared/pkg/families/secrets/gitleaks/config"
 	"github.com/openclarity/vmclarity/shared/pkg/families/vulnerabilities"
 )
 
+const (
+	TrivyTimeout       = 300
+	GrypeServerTimeout = 2 * time.Minute
+
+	GitleaksBinaryPath            = "GITLEAKS_BINARY_PATH"
+	ClamBinaryPath                = "CLAM_BINARY_PATH"
+	FreshclamBinaryPath           = "FRESHCLAM_BINARY_PATH"
+	AlternativeFreshclamMirrorURL = "ALTERNATIVE_FRESHCLAM_MIRROR_URL"
+	LynisInstallPath              = "LYNIS_INSTALL_PATH"
+	ChkrootkitBinaryPath          = "CHKROOTKIT_BINARY_PATH"
+	ExploitDBAddress              = "EXPLOIT_DB_ADDRESS"
+	TrivyServerAddress            = "TRIVY_SERVER_ADDRESS"
+	GrypeServerAddress            = "GRYPE_SERVER_ADDRESS"
```

Comment:
families logic should only receive a configuration struct from the outside, it should not be doing any loading of configurations itself, that is the responsibility of the CLI.

The CLI code uses viper + yaml, we should not be using environment variables like this anywhere in this flow, and if we want defaults then we should be handling it as part of loading that configuration. E.g. setting a default for rootkits.scannerconfig.chkrootkit.binarypath.

Reply:
I removed from families